[face-detector] Add isAvailableAsync method (#9680)
expo/expo: docs/pages/versions/unversioned/sdk/facedetector.md

// ...
```

### `FaceDetector.isAvailableAsync()`

> Returns whether the Face Detector API is enabled on the current device.

Reviewer: ```suggestion
Returns whether the Face Detection API is enabled on the current device.
```

#### Returns

> A `Promise` resolving to either `true` or `false`.

Reviewer: ```suggestion
Async `boolean`, indicating whether the Face Detection API is available on the current device. Currently this resolves `true` on iOS and Android only.
```

> Async `boolean`, indicating whether the Face Detection API is available on the current device. Currently this resolves `true` on iOS and Android only.

Reviewer 2: ```suggestion
Async `boolean`, indicating whether the Face Detection API is available on the current device. Currently this may resolve to `true` on iOS and Android only if the native code is linked properly.
```
